### PR TITLE
fix: subclassing data type in dialects

### PR DESCRIPTION
**Pull request by cyjake**
exports of the abstract data types should not be wrapped with invokable by default, which made subclassing like `class Postgres_DATE extends DataTypes.DATE {}` not working as expected because `DataTypes.DATE` was actually a Proxy instance.

This fix defers the invokable wrapping and adds more test cases regarding data types in dialects.

### CODE REVIEW

File: test/unit/drivers/postgres/attribute.test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  it('should support SERIAL', async function() {
+    const attribute=  new Attribute('id', {
+      type: INTEGER,
+      primaryKey: true,
+    });
+    assert.equal(attribute.toSqlString(), '"id" SERIAL PRIMARY KEY');
+  });
+
+  it('should support WITH or WITHOUT TIME ZONE', async function() {
```
Comment (JimmyDaddy) [inline test/unit/drivers/postgres/attribute.test.js:24]: Need a `WITHOUT TIME ZONE` test case.

File: test/unit/drivers/postgres/attribute.test.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    const attribute=  new Attribute('createdAt', {
+      type: DATE,
+    });
+    assert.equal(attribute.toSqlString(), '"created_at" TIMESTAMP WITH TIME ZONE');
```
Comment (JimmyDaddy) [inline test/unit/drivers/postgres/attribute.test.js:28]:
```suggestion
    assert.equal(attribute.toSqlString(), '"created_at" TIMESTAMP WITH TIME ZONE');
    const attribute1 =  new Attribute('createdAt', {
      type: new DATE(null, false),
    });
    assert.equal(attribute1.toSqlString(), '"created_at" TIMESTAMP WITHOUT TIME ZONE');
```

File: test/unit/drivers/postgres/attribute.test.js (hunk 1 above)
Comment (cyjake) [reply test/unit/drivers/postgres/attribute.test.js]: added

File: test/unit/drivers/postgres/attribute.test.js (hunk 2 above)
Comment (cyjake) [reply test/unit/drivers/postgres/attribute.test.js:28]: added a separated test

**Pull request merged**